### PR TITLE
Internal gameobject support

### PR DESCRIPTION
**Pull request by Jhonnyg**

### CODE REVIEW

File: defold-rive/src/comp_rive.cpp [hunk 1]
```diff
@@ -226,11 +217,161 @@ namespace dmRive
...
+        {
+            return false;
+        }
+
+        dmVMath::Vector3 inv_scale(1.0f, -1.0f, 1.0f);
+        dmVMath::Vector3 origin_position(-artboard->width()/2, artboard->height()/2, 0);
+
+        dmGameObject::SetScale(root_instance, inv_scale);
+        dmGameObject::SetPosition(root_instance, Point3(origin_position));
+        dmGameObject::SetParent(root_instance, rive_instance);
```
Comment (Jhonnyg) [inline defold-rive/src/comp_rive.cpp:312]: This is for conforming the coordinate system with Defold. Rive's coordinates have 0,0 at the top left corner of an artboard, but we want it to be centered around an image.

File: defold-rive/src/comp_rive.cpp [hunk 2]
```diff
@@ -259,7 +400,12 @@ namespace dmRive
         component->m_DoRender = 0;
         component->m_RenderConstants = 0;
 
-        // TODO: Create the bone hierarchy
+        if (!CreateGameobjects(world, component))
+        {
+            dmLogError("Failed to create game objects for paths in the rive model. Consider increasing collection max instances (collection.max_instances).");
```
Comment (JCash) [inline defold-rive/src/comp_rive.cpp:405]: The wording "Consider" means we're not sure what's going on.
This suggests that we're lacking a better error message earlier in the callstack?

**Review by JCash — APPROVED** [review]
Looks good

**Comment by JCash** [comment]
I'm guessing this PR is not relevant anymore, so I'm closing.

**Pull request closed without merging**